SpatialGDK/Source/SpatialGDK/Public/Schema/MigrationDiagnostic.h

```diff
@@ -0,0 +1,150 @@
+
+		FSpatialLoadBalancingHandler MigrationHandler(NetDriver);
+		FSpatialLoadBalancingHandler::EvaluateActorResult Result =
+			MigrationHandler.EvaluateSingleActor(BlockingActor, NetOwner, NewAuthWorkerId);
+		bool bCanMigrate = Result == FSpatialLoadBalancingHandler::EvaluateActorResult::Migrate;
+
+		Schema_AddBool(ResponseObject, SpatialConstants::MIGRATION_DIAGNOSTIC_EVALUATION_ID,
+					   Result == FSpatialLoadBalancingHandler::EvaluateActorResult::Migrate);
+		Schema_AddInt32(ResponseObject, SpatialConstants::MIGRATION_DIAGNOSTIC_DESTINATION_WORKER_ID, NewAuthWorkerId);
+		AddStringToSchema(ResponseObject, SpatialConstants::MIGRATION_DIAGNOSTIC_OWNER_ID, GetNameSafe(NetOwner));
+
+		return CommandResponse;
+	}
+
+	// Compare information from the worker that is blocked from migrating a hierarchy with the information from the authoritative
+	// worker over the actor that is blocking the migration and log the results.
+	static FString CreateMigrationDiagnosticLog(USpatialNetDriver* NetDriver, Schema_Object* ResponseObject, AActor* BlockingActor)
+	{
+		check(NetDriver != nullptr);
+		check(NetDriver->Connection != nullptr);
+		check(NetDriver->LockingPolicy != nullptr);
+
+		if (ResponseObject == nullptr)
+		{
+			return FString::Printf(TEXT("Migration diaganostic log failed as response was empty."));
+		}
+
+		VirtualWorkerId AuthoritativeWorkerId = Schema_GetInt32(ResponseObject, SpatialConstants::MIGRATION_DIAGNOSTIC_AUTHORITY_WORKER_ID);
+		Worker_EntityId BlockedEntityId = Schema_GetInt64(ResponseObject, SpatialConstants::MIGRATION_DIAGNOSTIC_ENTITY_ID);
+		bool bIsReplicated = GetBoolFromSchema(ResponseObject, SpatialConstants::MIGRATION_DIAGNOSTIC_REPLICATES_ID);
+		bool bHasAuthority = GetBoolFromSchema(ResponseObject, SpatialConstants::MIGRATION_DIAGNOSTIC_HAS_AUTHORITY_ID);
+		bool bIsLocked = GetBoolFromSchema(ResponseObject, SpatialConstants::MIGRATION_DIAGNOSTIC_LOCKED_ID);
+		bool bCanMigrate = GetBoolFromSchema(ResponseObject, SpatialConstants::MIGRATION_DIAGNOSTIC_EVALUATION_ID);
+		VirtualWorkerId DestinationWorkerId = Schema_GetInt32(ResponseObject, SpatialConstants::MIGRATION_DIAGNOSTIC_DESTINATION_WORKER_ID);
+		FString NetOwnerName = GetStringFromSchema(ResponseObject, SpatialConstants::MIGRATION_DIAGNOSTIC_OWNER_ID);
+
+		FString Reason = FString::Printf(TEXT("Originating worker %i does not have authority of blocking actor. "),
+										 NetDriver->VirtualWorkerTranslator->GetLocalVirtualWorkerId());
+
+		if (bHasAuthority)
+		{
+			Reason.Append(FString::Printf(TEXT("Worker %i has authority of blocking actor. "), AuthoritativeWorkerId));
+		}
+
+		if (!IsValid(BlockingActor))
+		{
+			Reason.Append(FString::Printf(TEXT("Blocking actor is not valid. ")));
+		}
+		else if (!IsValid(BlockingActor->GetOwner()))
+		{
+			Reason.Append(FString::Printf(TEXT("Blocking actor owner is not valid. ")));
+		}
+
+		if (IsValid(BlockingActor) && BlockingActor->GetIsReplicated() && !bIsReplicated)
```

Comment:
These should be cascading else if. If any of the conditions are not right, they would not proceed to the next step.
If it's not replicated, it won't migrate.
If the remote owner is wrong, the destination won't matter, etc...

Reply:
Great, thanks that's useful to know.